src/js/other-websites/inpage_menu.js

```diff
@@ -193,26 +276,64 @@ function applySearchFilter(query) {
     }
   });
 
-  // Add #/# labels to search filter
+  // Set #/# labels inside search bar to show results count
   const searchFilterTotal = document.querySelector(".js-filter-masks-total");
   const searchFilterVisible = document.querySelector(
     ".js-filter-masks-visible"
   );
 
-  searchFilterVisible.textContent = maskResults.length;
+  searchFilterVisible.textContent = maskSearchResults.filter((maskResult) => !maskResult.classList.contains("is-hidden")).length;
+  searchFilterTotal.textContent = maskSearchResults.length;
+}
+
+function checkIfAnyMasksWereGeneratedOnCurrentWebsite(masks, domain) {
+  return masks.some((mask) => {
+    return domain === mask.generated_for;
+  });
+}
+
+function hasMaskBeenUsedOnCurrentSite(mask, domain) {
+  const domainList = mask.used_on;
+
+  // Short circuit out if there's no used_on entry
+  if (domainList === null || domainList === "" ||  domainList === undefined) { return false; }
+
+  const usedOnDomains = domainList.split(",");
+
+  // Domain already exists in used_on field. Just return the list!
+  if (usedOnDomains.includes(domain)) {
+    return true;
+  }
+
+  // No match found! 
+  return false;
+}
+
+function haveMasksBeenUsedOnCurrentSite(masks, domain) {
+  return masks.some(mask => {
+    
+    const domainList = mask.used_on;
+
+    // Short circuit out if there's no used_on entry
+    if (domainList === null || domainList === "" ||  domainList === undefined) { return false; }
+
+    const usedOnDomains = domainList.split(",");
+
+    // Domain already exists in used_on field. Just return the list!
+    if (usedOnDomains.includes(domain)) {
+      return true;
+    }
```

Comment:
suggestion (non-blocking): again, maybe we can call `includes()` directly on the string here?

Reply:
Revised in d5e6221fe717faa6809d549a9e0b85259f67a063